### PR TITLE
Added Split-Cloud, a Handwired Keyboard

### PR DESCRIPTION
**Pull request by coolsa**
## Description

Add a "new" 5x12 ortholinear handwired split keyboard, with big inspiration from the Nyquist and the Preonic.

The default keymap has some modifications from the main Preonic layout, the major difference is that this supports stenography by default.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/split_cloud/config.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+// #define QMK_SPEAKER C6
+
+// #define AUDIO_VOICES
+// #define AUDIO_PIN C6
+
+// #define BACKLIGHT_PIN B7
+
+/* define if matrix has ghost */
+// #define MATRIX_HAS_GHOST
+
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/config.h:39]:
```suggestion
```

File: keyboards/handwired/split_cloud/info.json [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        },
+        "bootmagic": {
+            "matrix": [5, 5]
+        }
+    },
+    "stenography": {
+        "enabled": true,
+        "protocol": "all"
+    }
+    "debounce": 5,
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:38]:
```suggestion
```

File: keyboards/handwired/split_cloud/info.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
+    "community_layouts": ["ortho_5x12"],
+    "layout_aliases": {
+        "LAYOUT": "LAYOUT_ortho_5x12",
+        "LAYOUT_preonic_grid": "LAYOUT_ortho_5x12"
+    },
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:45]:
```suggestion
```

File: keyboards/handwired/split_cloud/info.json [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                {"matrix": [0, 3], "x": 3, "y": 0},
+                {"matrix": [0, 4], "x": 4, "y": 0},
+                {"matrix": [0, 5], "x": 5, "y": 0},
+
+                {"matrix": [5, 5], "x": 6, "y": 0},
+                {"matrix": [5, 4], "x": 7, "y": 0},
+                {"matrix": [5, 3], "x": 8, "y": 0},
+                {"matrix": [5, 2], "x": 9, "y": 0},
+                {"matrix": [5, 1], "x": 10, "y": 0},
+                {"matrix": [5, 0], "x": 11, "y": 0},
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:61]:
```suggestion
                {"matrix": [5, 5], "x": 7, "y": 0},
                {"matrix": [5, 4], "x": 8, "y": 0},
                {"matrix": [5, 3], "x": 9, "y": 0},
                {"matrix": [5, 2], "x": 10, "y": 0},
                {"matrix": [5, 1], "x": 11, "y": 0},
                {"matrix": [5, 0], "x": 12, "y": 0},
```

File: keyboards/handwired/split_cloud/info.json [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                {"matrix": [1, 3], "x": 3, "y": 1},
+                {"matrix": [1, 4], "x": 4, "y": 1},
+                {"matrix": [1, 5], "x": 5, "y": 1},
+
+                {"matrix": [6, 5], "x": 6, "y": 1},
+                {"matrix": [6, 4], "x": 7, "y": 1},
+                {"matrix": [6, 3], "x": 8, "y": 1},
+                {"matrix": [6, 2], "x": 9, "y": 1},
+                {"matrix": [6, 1], "x": 10, "y": 1},
+                {"matrix": [6, 0], "x": 11, "y": 1},
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:75]:
```suggestion
                {"matrix": [6, 5], "x": 7, "y": 1},
                {"matrix": [6, 4], "x": 8, "y": 1},
                {"matrix": [6, 3], "x": 9, "y": 1},
                {"matrix": [6, 2], "x": 10, "y": 1},
                {"matrix": [6, 1], "x": 11, "y": 1},
                {"matrix": [6, 0], "x": 12, "y": 1},
```

File: keyboards/handwired/split_cloud/info.json [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                {"matrix": [2, 3], "x": 3, "y": 2},
+                {"matrix": [2, 4], "x": 4, "y": 2},
+                {"matrix": [2, 5], "x": 5, "y": 2},
+
+                {"matrix": [7, 5], "x": 6, "y": 2},
+                {"matrix": [7, 4], "x": 7, "y": 2},
+                {"matrix": [7, 3], "x": 8, "y": 2},
+                {"matrix": [7, 2], "x": 9, "y": 2},
+                {"matrix": [7, 1], "x": 10, "y": 2},
+                {"matrix": [7, 0], "x": 11, "y": 2},
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:89]:
```suggestion
                {"matrix": [7, 5], "x": 7, "y": 2},
                {"matrix": [7, 4], "x": 8, "y": 2},
                {"matrix": [7, 3], "x": 9, "y": 2},
                {"matrix": [7, 2], "x": 10, "y": 2},
                {"matrix": [7, 1], "x": 11, "y": 2},
                {"matrix": [7, 0], "x": 12, "y": 2},
```

File: keyboards/handwired/split_cloud/info.json [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                {"matrix": [3, 3], "x": 3, "y": 3},
+                {"matrix": [3, 4], "x": 4, "y": 3},
+                {"matrix": [3, 5], "x": 5, "y": 3},
+
+                {"matrix": [8, 5], "x": 6, "y": 3},
+                {"matrix": [8, 4], "x": 7, "y": 3},
+                {"matrix": [8, 3], "x": 8, "y": 3},
+                {"matrix": [8, 2], "x": 9, "y": 3},
+                {"matrix": [8, 1], "x": 10, "y": 3},
+                {"matrix": [8, 0], "x": 11, "y": 3},
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:103]:
```suggestion
                {"matrix": [8, 5], "x": 7, "y": 3},
                {"matrix": [8, 4], "x": 8, "y": 3},
                {"matrix": [8, 3], "x": 9, "y": 3},
                {"matrix": [8, 2], "x": 10, "y": 3},
                {"matrix": [8, 1], "x": 11, "y": 3},
                {"matrix": [8, 0], "x": 12, "y": 3},
```

File: keyboards/handwired/split_cloud/info.json [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+                {"matrix": [4, 3], "x": 3, "y": 4},
+                {"matrix": [4, 4], "x": 4, "y": 4},
+                {"matrix": [4, 5], "x": 5, "y": 4},
+
+                {"matrix": [9, 5], "x": 6, "y": 4},
+                {"matrix": [9, 4], "x": 7, "y": 4},
+                {"matrix": [9, 3], "x": 8, "y": 4},
+                {"matrix": [9, 2], "x": 9, "y": 4},
+                {"matrix": [9, 1], "x": 10, "y": 4},
+                {"matrix": [9, 0], "x": 11, "y": 4}
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:117]:
```suggestion
                {"matrix": [9, 5], "x": 7, "y": 4},
                {"matrix": [9, 4], "x": 8, "y": 4},
                {"matrix": [9, 3], "x": 9, "y": 4},
                {"matrix": [9, 2], "x": 10, "y": 4},
                {"matrix": [9, 1], "x": 11, "y": 4},
                {"matrix": [9, 0], "x": 12, "y": 4}
```

File: keyboards/handwired/split_cloud/split_cloud.c [hunk 9]
```diff
@@ -0,0 +1,18 @@
+/*
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/split_cloud.c:1]: This file can be deleted.

File: keyboards/handwired/split_cloud/readme.md [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+![Split-Cloud](https://raw.githubusercontent.com/coolsa/cloud-keyboard/main/images/finalProduct.jpg)
+
+* Keyboard Maintainer: [Cloud Chagnon](https://github.com/coolsa)
+* Hardware Supported: Pro Micro
+* Hardware Availability: [Repository](https://github.com/coolsa/split-cloud)
+
+Make example for this keyboard:
+
+    make handwired/split_cloud:default:avrdude-split-left
+    make handwired/split_cloud:default:avrdude-split-right
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/readme.md:14]:
```suggestion
    make handwired/split_cloud:default
```

File: keyboards/handwired/split_cloud/keymaps/default/config.h [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+/* enable advanced MIDI features:
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+// #define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+// #define MIDI_TONE_KEYCODE_OCTAVES 2
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/config.h:38]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,353 @@
+/* Copyright 2015-2021 Jack Humbert
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:1]: Please keep default keymaps simple. Remove anything you aren't using.

File: keyboards/handwired/split_cloud/keymaps/default/rules.mk [hunk 13]
```diff
@@ -0,0 +1,5 @@
+ifeq ($(strip $(AUDIO_ENABLE)), yes)
+    SRC += muse.c
+endif
+STENO_ENABLE = yes
+#EXTRAKEY_ENABLE = no
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/rules.mk:5]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/readme.md [hunk 14]
```diff
@@ -0,0 +1,3 @@
+# The default Split Cloud layout - largely based on the Preonic's and the Planck's steno layout.
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/readme.md:1]: This file can be deleted.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/split_cloud/readme.md (hunk 10 above)
Comment (coolsa) [reply keyboards/handwired/split_cloud/readme.md:18]: I altered the readme to have this, but also added a few lines describing the usage of handedness by eeprom. Let me know if my changes are good, or if I should trim that section out entirely.

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,259 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "keymap_steno.h"
+
+extern keymap_config_t keymap_config;
+
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:21]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,259 @@
...
+                writePinLow(E6);
+#endif
+            } else {
+                unregister_code(KC_RSFT);
+#ifdef __AVR__
+                writePinHigh(E6);
+#endif
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:244]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 17]
```diff
@@ -0,0 +1,259 @@
...
+        case PLOVER:
+            if (!record->event.pressed) {
+                layer_on(_PLOVER);
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:250]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 18]
```diff
@@ -0,0 +1,259 @@
...
+        case EXT_PLV:
+            if (record->event.pressed) {
+                layer_off(_PLOVER);
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:256]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 19]
```diff
@@ -0,0 +1,259 @@
...
+        case RAISE:
+            if (record->event.pressed) {
+                layer_on(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            } else {
+                layer_off(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:224]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,259 @@
...
+        case LOWER:
+            if (record->event.pressed) {
+                layer_on(_LOWER);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            } else {
+                layer_off(_LOWER);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:214]:
```suggestion
```

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,259 @@
...
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+            break;
+        case COLEMAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_COLEMAK);
+            }
+            return false;
+            break;
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:198]:
```suggestion
```

File: keyboards/handwired/split_cloud/info.json [hunk 22]
```diff
@@ -0,0 +1,116 @@
...
+    },
+    "diode_direction": "ROW2COL",
+    "matrix_pins": {
+        "cols": ["B6", "B2", "B3", "B1", "F7", "F6"],
+        "rows": ["C6", "D7", "E6", "B4", "B5"]
+    },
+    "bootmagic": {
+        "enabled": true,
+        "matrix": [0, 0]
+    },
```
Comment (fauxpark) [inline keyboards/handwired/split_cloud/info.json:20]:
```suggestion
    },
    "features": {
        "extrakey": true,
        "nkro": true
    },
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by coolsa** [comment]
I've made the suggested changes. I did also move EE_HANDS from config.h to info.json, but ran into some issues. There is a PR open to fix it though.

File: keyboards/handwired/split_cloud/info.json [hunk 23]
```diff
@@ -0,0 +1,120 @@
...
+        "force_nkro": true
+    },
+    "diode_direction": "ROW2COL",
+    "matrix_pins": {
+        "cols": ["B6", "B2", "B3", "B1", "F7", "F6"],
+        "rows": ["C6", "D7", "E6", "B4", "B5"]
+    },
+    "bootmagic": {
+        "enabled": true,
+        "matrix": [0, 0]
```
Comment (dunk2k) [inline keyboards/handwired/split_cloud/info.json:19]: default matrix co-ordinates for bootmagic
```suggestion
        "enabled": true
```

File: keyboards/handwired/split_cloud/info.json [hunk 24]
```diff
@@ -0,0 +1,120 @@
...
+        "bootmagic": {
+            "matrix": [5, 5]
+        }
+    },
+    "stenography": {
+        "enabled": true,
+        "protocol": "all"
+    }
+    "processor": "atmega32u4",
+    "bootloader": "caterina",
```
Comment (dunk2k) [inline keyboards/handwired/split_cloud/info.json:43]:
```suggestion
    "development_board": "promicro",
```

File: keyboards/handwired/split_cloud/keymaps/default/rules.mk
Comment (dunk2k) [inline keyboards/handwired/split_cloud/keymaps/default/rules.mk:1]: this file can be removed as stenography is enabled in `info.json`

File: keyboards/handwired/split_cloud/readme.md [hunk 25]
```diff
@@ -0,0 +1,22 @@
+# Split-Cloud handwired keyboard
+
+The Split-Cloud is a split ortholinear 5x12 layout, forked from the Preonic and Nyquist keyboard models. The default layout adds extra options for some added game compatibility, two control keys for QEmu, and adds a steno layer based on the Planck example.
+
+![Split-Cloud](https://raw.githubusercontent.com/coolsa/cloud-keyboard/main/images/finalProduct.jpg)
```
Comment (dunk2k) [inline keyboards/handwired/split_cloud/readme.md:5]: Please re-host this image on imgur

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/handwired/split_cloud/keymaps/default/keymap.c [hunk 26]
```diff
@@ -0,0 +1,246 @@
...
+            return false;
+        case RAISE:
+            if (record->event.pressed) {
+                layer_on(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            } else {
+                layer_off(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
```
Comment (drashna) [inline keyboards/handwired/split_cloud/keymaps/default/keymap.c:233]: These do not need custom keycodes anymore, as this functionality has been added to core: 
https://docs.qmk.fm/#/feature_tri_layer

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/handwired/split_cloud/readme.md [hunk 27]
```diff
@@ -0,0 +1,22 @@
+# Split-Cloud handwired keyboard
+
+The Split-Cloud is a split ortholinear 5x12 layout, forked from the Preonic and Nyquist keyboard models. The default layout adds extra options for some added game compatibility, two control keys for QEmu, and adds a steno layer based on the Planck example.
+
+![Split-Cloud](https://i.imgur.com/QoHAFeX.jpg)
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/handwired/split_cloud/readme.md:5]: This suggestion uses a smaller, more github markdown friendly version of the image, that is still decently large.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**